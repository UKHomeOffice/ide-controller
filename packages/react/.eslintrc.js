// @todo add vs code format on save wiki/ readme on github
module.exports = {
  env: {
    browser: true,
    es2020: true,
    node: true,
    jest: true,
  },
  extends: ['plugin:react/recommended', 'airbnb', 'prettier'],
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 11,
    sourceType: 'module',
  },
  plugins: ['react', 'prettier'],
  ignorePatterns: ['/src/opencv'],
  rules: {
    quotes: ['error', 'single'],
    'prettier/prettier': [
      'error',
      {
        trailingComma: 'es5',
        singleQuote: true,
        printWidth: 80,
      },
    ],
  },
};
