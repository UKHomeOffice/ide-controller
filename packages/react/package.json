{
  "name": "@ide-controller/react",
  "version": "1.6.0",
  "private": true,
  "homepage": "./",
  "author": "Viable Data",
  "scripts": {
    "dev": "BROWSER=none CI=true react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --watchAll=false",
    "test:watch": "react-scripts test",
    "eject": "react-scripts eject",
    "lint": "npx eslint --ext .js,.jsx .",
    "lint:fix": "eslint --ext .js,.jsx . --fix"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "dependencies": {
    "@tensorflow-models/posenet": "^2.2.1",
    "@tensorflow/tfjs": "^2.7.0",
    "govuk-frontend": "^3.8.0",
    "prop-types": "^15.7.2",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "^3.4.3"
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.3.2",
    "@testing-library/user-event": "^7.1.2",
    "eslint-config-airbnb": "^18.2.0",
    "eslint-config-prettier": "^6.11.0",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-prettier": "^3.1.4",
    "eslint-plugin-react": "^7.21.5",
    "jest-canvas-mock": "^2.3.0",
    "node-sass": "^4.14.1",
    "prettier": "^2.1.1"
  }
}
