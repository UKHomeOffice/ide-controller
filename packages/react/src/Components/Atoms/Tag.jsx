// Golbal imports
import PropTypes from 'prop-types';
import React from 'react';

const Tag = ({ statusText, status }) => {
  return (
    <strong className={`govuk-tag tag-state--${status}`}>{statusText}</strong>
  );
};

Tag.propTypes = {
  status: PropTypes.string.isRequired,
  statusText: PropTypes.string.isRequired,
};

export default Tag;
