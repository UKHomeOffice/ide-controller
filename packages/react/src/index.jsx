// Global imports
import React from 'react';
import ReactDOM from 'react-dom';
import { initAll } from 'govuk-frontend';
import 'govuk-frontend/govuk/all.scss';

// Local imports
import App from './App';
import './Components/Style/atoms.scss';
import './Components/Style/global.scss';
import './Components/Style/molecules.scss';
import './helpers/ipcMainEvents';

document.body.className = document.body.className
  ? `${document.body.className} js-enabled`
  : 'js-enabled';

initAll();

ReactDOM.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  document.getElementById('root')
);
