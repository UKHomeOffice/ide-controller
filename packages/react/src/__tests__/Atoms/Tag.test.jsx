// Global imports
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// Local imports
import Tag from '../../Components/Atoms/Tag';

describe('<Tag />', () => {
  it('renders and match the snapshot', () => {
    const { asFragment } = render(
      <Tag tagText="tagText" tagStatus="neutral" />
    );

    expect(asFragment).toMatchSnapshot();
  });

  it('has text content as tagText', () => {
    const tagText = 'testTag';
    render(<Tag tagText={tagText} tagStatus="neutral" />);
    const tag = screen.getByTestId('atoms-tag');

    expect(tag).toHaveTextContent(tagText);
  });

  it('has className depended on tagStatus prop', () => {
    const testStatus = 'neutral';
    render(<Tag tagText="testTag" tagStatus={testStatus} />);
    const tag = screen.getByTestId('atoms-tag');

    expect(tag).toHaveClass(`bg--${testStatus}`);
  });
});
