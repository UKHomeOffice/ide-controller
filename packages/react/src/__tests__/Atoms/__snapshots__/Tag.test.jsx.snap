// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> renders and match the snapshot 1`] = `[Function]`;
